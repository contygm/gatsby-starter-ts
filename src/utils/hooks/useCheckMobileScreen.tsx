import { useEffect, useState } from 'react';

/**
 * a quick hook to check if the screen is mobile or tablet size
 * @function useCheckMobileScreen
 * @returns {boolean}
 */
const useCheckMobileScreen = ( breakpoint = 1216 ) => {
    const hasWindow = typeof window !== 'undefined';

    if (hasWindow) {
        const [width, setWidth] = useState(window.innerWidth);

        const handleWindowSizeChange = () => {
            setWidth(window.innerWidth);
        };

        useEffect(() => {
            window.addEventListener('resize', handleWindowSizeChange);
            return () => {
                window.removeEventListener('resize', handleWindowSizeChange);
            };
        }, []);

        return width < breakpoint;
    }
    return false;
};

export default useCheckMobileScreen;
