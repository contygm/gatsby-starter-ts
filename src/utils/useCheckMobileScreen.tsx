import { useEffect, useState } from 'react';

const useCheckMobileScreen = () => {
    const hasWindow = typeof window !== "undefined";

    if (hasWindow) {
        const [width, setWidth] = useState(window.innerWidth);

        const handleWindowSizeChange = () => {
            setWidth(window.innerWidth);
        };

        useEffect(() => {
            window.addEventListener('resize', handleWindowSizeChange);
            return () => {
                window.removeEventListener('resize', handleWindowSizeChange);
            };
        }, []);

        return width <= 1023;
    }
    return false;
};

export default useCheckMobileScreen;
