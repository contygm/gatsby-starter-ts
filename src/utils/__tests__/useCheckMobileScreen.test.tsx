import { act, render } from '@testing-library/react';
import React from 'react';
import useCheckMobileScreen from '../useCheckMobileScreen';

const resizeWindow = (x: number) => {
    window.innerWidth = x;
    window.dispatchEvent(new Event('resize'));
}

function TestComponent() {
	if (useCheckMobileScreen()) {
	  return <div>SMALL SCREEN</div>;
	}
	return <div>BIG SCREEN</div>;
}

describe('useCheckBigScreen', () => {
	beforeEach(() => {
        resizeWindow(900)

    })

    it('return siteMetadata', () => {
		const { asFragment } = render(<TestComponent />);
        expect(asFragment()).toMatchSnapshot('small');
		
		act(() => {
			resizeWindow(1216)
		});
		expect(asFragment()).toMatchSnapshot('big');
    });
});