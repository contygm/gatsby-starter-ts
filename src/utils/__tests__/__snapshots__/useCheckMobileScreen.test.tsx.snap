// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useCheckBigScreen return siteMetadata: big 1`] = `
<DocumentFragment>
  <div>
    BIG SCREEN
  </div>
</DocumentFragment>
`;

exports[`useCheckBigScreen return siteMetadata: small 1`] = `
<DocumentFragment>
  <div>
    SMALL SCREEN
  </div>
</DocumentFragment>
`;
